### PR TITLE
Feature web support

### PR DESCRIPTION
**Pull request by theiskaa**
### Description: 
I enabled the web to help you with the time of building/deployment the web app. If you agree and building/deploying a web app then other guys can take look directly at the project, so without cloning repo and etc. However, it will make it easy for people, who read your posts.

Again thanks for the design patterns series, you're a legend! ❤️ 

### Result: 

https://user-images.githubusercontent.com/59066341/118487888-39bc0700-b72c-11eb-9b0e-bda38ae601c9.mov

### CODE REVIEW

File: lib/widgets/design_patterns/strategy/order/order_buttons/order_button.dart [hunk 1]
```diff
@@ -32,20 +32,34 @@ class OrderButton extends StatelessWidget {
...
-          : CupertinoButton(
-              color: Colors.black,
-              onPressed: onPressed,
-              child: child,
-            ),
+      child: getRightButton(child),
     );
   }
+
+  Widget getRightButton(Widget child) {
```
Comment (mkobuolys) [inline lib/widgets/design_patterns/strategy/order/order_buttons/order_button.dart:39]: This function is not needed.

File: lib/widgets/design_patterns/strategy/order/order_buttons/order_button.dart [hunk 2]
```diff
@@ -32,20 +32,34 @@ class OrderButton extends StatelessWidget {
 
     return Padding(
       padding: const EdgeInsets.all(4.0),
-      child: Platform.isAndroid
```
Comment (mkobuolys) [inline lib/widgets/design_patterns/strategy/order/order_buttons/order_button.dart:35]: Extend this check with: `kIsWeb || Platform.isAndroid`.

File: lib/widgets/platform_specific/platform_back_button.dart [hunk 3]
```diff
@@ -12,11 +12,22 @@ class PlatformBackButton extends StatelessWidget {
...
-      icon: Icon(
-        Platform.isAndroid ? Icons.arrow_back : Icons.arrow_back_ios,
-      ),
+      icon: Icon(getValidIcon),
       color: color,
       onPressed: () => Navigator.pop(context),
     );
   }
+
+  IconData get getValidIcon {
```
Comment (mkobuolys) [inline lib/widgets/platform_specific/platform_back_button.dart:21]: This function is not needed.

File: lib/widgets/platform_specific/platform_back_button.dart [hunk 4]
```diff
@@ -12,11 +12,22 @@ class PlatformBackButton extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return IconButton(
-      icon: Icon(
-        Platform.isAndroid ? Icons.arrow_back : Icons.arrow_back_ios,
```
Comment (mkobuolys) [inline lib/widgets/platform_specific/platform_back_button.dart:16]: Extend this check with: `kIsWeb || Platform.isAndroid`.

File: lib/widgets/platform_specific/platform_button.dart [hunk 5]
```diff
@@ -18,22 +18,33 @@ class PlatformButton extends StatelessWidget {
...
-            )
-          : CupertinoButton(
-              color: Colors.black,
-              onPressed: onPressed,
-              child: Text(text, textAlign: TextAlign.center),
-            ),
+    return Padding(padding: const EdgeInsets.all(4.0), child: getRightButton);
+  }
+
+  Widget get getRightButton {
```
Comment (mkobuolys) [inline lib/widgets/platform_specific/platform_button.dart:24]: This function is not needed.

File: lib/widgets/platform_specific/platform_button.dart [hunk 6]
```diff
@@ -18,22 +18,33 @@ class PlatformButton extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Padding(
-      padding: const EdgeInsets.all(4.0),
-      child: Platform.isAndroid
```
Comment (mkobuolys) [inline lib/widgets/platform_specific/platform_button.dart:23]: Extend this check with: `kIsWeb || Platform.isAndroid`.

File: web/index.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    it to work correctly.
+
+    Fore more details:
+    * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base
+  -->
+  <base href="/">
+
+  <meta charset="UTF-8">
+  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
+  <meta name="description" content="A new Flutter project.">
```
Comment (mkobuolys) [inline web/index.html:18]: Update the content to "A Flutter Design Patterns Library."

File: web/manifest.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "flutter_design_patterns",
+    "short_name": "flutter_design_patterns",
+    "start_url": ".",
+    "display": "standalone",
+    "background_color": "#0175C2",
+    "theme_color": "#0175C2",
+    "description": "A new Flutter project.",
```
Comment (mkobuolys) [inline web/manifest.json:8]: Update description to "A Flutter Design Patterns Library."

**Review by mkobuolys — CHANGES_REQUESTED** [review]
Looking good, just some details to adjust and some code to revert/improve.

**Comment by theiskaa** [comment]
@mkobuolys I applied all things that you were commented on. So you can re-review PR

**Review by mkobuolys — APPROVED** [review]
LGTM, thanks for your submission! Next step - making this actually look good on the Web. However, that's on my side 😉

**Comment by theiskaa** [comment]
Thanks! I can't wait to see the project on the web. 🚀

**Pull request merged**